slick-codegen/src/main/scala/slick/codegen/AbstractSourceCodeGenerator.scala

```diff
@@ -57,14 +57,42 @@ abstract class AbstractSourceCodeGenerator(model: m.Model)
       else compoundValue(types)
     }
 
-    def compoundValue(values: Seq[String]): String = {
+    //build and nest tuples if > 22 elements
+    def tupleBuilder(values: Seq[String] , prefix: String): String = {
+      def mkTuple[T](l: Seq[T], _prefix: String) = l.map{ _prefix + _ }.mkString("(" , " ," , ")")
+      if(values.size <= 22) 
+        mkTuple(values,"") 
+      else 
+        mkTuple( values.grouped(22).toList.map {mkTuple(_,prefix)} , ""  )
+    }
+
+    def tupleFactoryBuilder(columns: Seq[Column]): String = {
+        if(columns.size <= 22 ) s"${TableClass.elementType}.tupled"
+        else{
+          val count = (columns.size / 22.0).ceil.toInt
+          val tuples = Range(0,count).map{ "t" + _ }
+          s"""{case ${tuples.mkString("(",",",")")} => ${TableClass.elementType}""" + tuples.map {
+            tple => {
+              val mod = if(tuples.size == 1 ) columns.size 
+                        else if(tuples.indexOf(tple) == 0 && columns.size > 22) 22  
+                        else columns.size%(tuples.indexOf(tple) * 22)
+              val entries = if( mod == 0 ) 22 else mod   
+              Range(0,entries).map{ s"${tple}." + tuple(_) }
+            }
+          }.flatten.mkString("(" , " ," , ")") + "}" 
+        }
+    }
+
+    def compoundValue(values: Seq[String]) = compoundValue(values , "")
+    def compoundValue(values: Seq[String] , prefix: String): String = {
       if(hlistEnabled) values.mkString(" :: ") + " :: HNil"
       else if (values.size == 1) values.head
-      else if(values.size <= 22) s"""(${values.mkString(", ")})"""
-      else throw new Exception("Cannot generate tuple for > 22 columns, please set hlistEnable=true or override compound.")
+      else tupleBuilder(values, prefix)
     }
 
-    def factory   = if(columns.size == 1) TableClass.elementType else s"${TableClass.elementType}.tupled"
+    def factory   = if(columns.size == 1) TableClass.elementType 
+                    else if( hlistEnabled ) columns.mkString("(" , " ::" , " HNil")
```

Comment:
this looks wrong. Where is this "(" closed?

(Also `::` is usually written with a space before AND after, unlike `,` which is usually written with ONLY a space AFTER.)